pytket/pytket/qasm/qasm.py

```diff
@@ -854,35 +854,52 @@ def _get_optype_and_params(op: Op) -> Tuple[OpType, Optional[List[float]]]:
     return (optype, params)
 
 
+def hqs_header(header: str) -> bool:
+    if header == "hqslib1":
+        return True
+
+    if header == "hqslib1_dev":
+        return True
+
+    return False
+
+
 def circuit_to_qasm_io(
     circ: Circuit,
     stream_out: TextIO,
     header: str = "qelib1",
     include_gate_defs: Optional[Set[str]] = None,
 ) -> None:
     """A method to generate a qasm text stream from a tket Circuit"""
-    if (
-        any(
-            circ.n_gates_of_type(typ)
-            for typ in (
-                OpType.RangePredicate,
-                OpType.MultiBit,
-                OpType.ExplicitPredicate,
-                OpType.ExplicitModifier,
-                OpType.SetBits,
-                OpType.CopyBits,
-                OpType.ClassicalExpBox,
-            )
+    if any(
+        circ.n_gates_of_type(typ)
+        for typ in (
+            OpType.RangePredicate,
+            OpType.MultiBit,
+            OpType.ExplicitPredicate,
+            OpType.ExplicitModifier,
+            OpType.SetBits,
+            OpType.CopyBits,
+            OpType.ClassicalExpBox,
         )
-        and header != "hqslib1"
-    ):
+    ) and (not hqs_header(header)):
         raise QASMUnsupportedError(
             "Complex classical gates only supported with hqslib1."
         )
     if include_gate_defs is None:
         include_gate_defs = {"measure", "reset", "barrier"}
         include_gate_defs.update(_load_include_module(header, False, True).keys())
-        stream_out.write('OPENQASM 2.0;\ninclude "{}.inc";\n\n'.format(header))
+
+        split_header = header.split("_dev")
+        if len(split_header) > 1:
+            stream_out.write(
+                'OPENQASM 2.0;\ninclude "{}_dev.inc";\ninclude "{}.inc";\n\n'.format(
```

Comment:
Are you sure we need to include both? According to our internal wiki you only need to include the "dev" one. That would remove the need for this complex logic altogether, I think.

Reply:
From the examples I have seen it is required to have both, can change that I wanted.